drivers/cluster/offload_snitchCluster.c

```diff
@@ -0,0 +1,196 @@
+// Copyright 2024 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Moritz Scherer <scheremo@iis.ee.ethz.ch>
+// Philip Wiese <wiesep@iis.ee.ethz.ch>
+
+#include "soc.h"
+#include "offload_snitchCluster.h"
+
+#include <stdint.h>
+#include <stddef.h>
+
+// Persistent trampoline function pointer for each core
+void (*_trampoline_function[NUM_CLUSTER_CORES])(void *) = {NULL};
+
+// Peristent argument storage for the trampoline function
+void *_trampoline_args[NUM_CLUSTER_CORES] = {NULL};
+
+// Persistant stack pointer storage for each core
+void *_trampoline_stack[NUM_CLUSTER_CORES] = {NULL};
```

Comment:
FUTURE: Thinking of mixed-cluster or mixed-inter-cluster-ISA chimeras, we might want to have this explicitly only be for snitch cores.

Reply:
I opened a feature issue #16 for this.